### PR TITLE
fixed the upload of image for docker extension

### LINKED ISSUE #5747: [UI] Docker Extension Upload of image in browse compose app
#### Current Behavior

In Docker Extension , when user upload any other file type like png it still show the message uploaded. 

#### Desired Behavior

Disallow the upload of images .  

#### Screenshots/Logs

image

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources

- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by harkiratsm**
Signed-off-by: harkiratsm <multaniharry714@gmail.com>

**Description**

This PR fixes #5747

**Notes for Reviewers**

When a user try top upload image the error msg is displayed . 

image

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by warunicorn19 — COMMENTED** [review]
Seems like a one off situation, how exactly did you get this?? 
when I uploaded my compose app, it happened successfully. 
Screenshot 2022-07-03 at 7 34 20 PM

I installed the extension from docker-desktop itself.

**Comment by harkiratsm** [comment]
@warunicorn19  Try to upload a PNG file , then it will show uploaded , but instead it should throw a error . This PR fixes that issue .

File: install/docker-extension/ui/src/components/ExtensionComponent/ExtensionComponent.js [hunk 1]
```diff
@@ -230,19 +231,25 @@ const ExtensionsComponent = () => {
...
-      fetch(proxyUrl + "/api/application", {
-        method: "POST",
-        headers: { "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8" },
-        body,
-      })
-        .then((res) => {
-          console.log(res);
-          window.ddClient.desktopUI.toast.success("Compose file has been uploaded with name: " + name);
+      if(!(type.includes(".yaml")||type.includes(".yml"))){
+        window.ddClient.desktopUI.toast.error("Some error occured while uploading the compose file. ")
```
Comment (sudo-NithishKarthik) [inline install/docker-extension/ui/src/components/ExtensionComponent/ExtensionComponent.js:235]: adding a `return` statement would prevent it from moving further and executing unwanted lines of code.

Comment (sudo-NithishKarthik) [reply install/docker-extension/ui/src/components/ExtensionComponent/ExtensionComponent.js:235]: it is a better way of expressing the intention as well. And you don't have to use `else` 😉

Comment (sudo-NithishKarthik) [reply install/docker-extension/ui/src/components/ExtensionComponent/ExtensionComponent.js:235]: Nn this case, it doesn't really matter.

File: install/docker-extension/ui/src/components/ExtensionComponent/ExtensionComponent.js [hunk 2]
```diff
@@ -222,6 +222,7 @@ const ExtensionsComponent = () => {
   const handleImport = () => {
     const file = document.getElementById("upload-button").files[0];
     // Create a reader
+    let type=String(file.name)
```
Comment (sudo-NithishKarthik) [inline install/docker-extension/ui/src/components/ExtensionComponent/ExtensionComponent.js:225]: use `let` only when you think this is going to change.

**Review by sudo-NithishKarthik — CHANGES_REQUESTED** [review]
Thank you @harkiratsm 😄 . Some nitpicks.

File: install/docker-extension/ui/src/components/ExtensionComponent/ExtensionComponent.js (hunk 1 above)
Comment (harkiratsm) [reply install/docker-extension/ui/src/components/ExtensionComponent/ExtensionComponent.js:235]: Thanks 😄

**Review by sudo-NithishKarthik — APPROVED** [review]
LGTM 🎉

**Pull request merged**